enarx-keep-sgx-shim/src/start.S

```diff
@@ -163,6 +191,7 @@ syscall:
 
     xor     %rcx,                   %rcx            # Clear %rcx
     zerof   %rcx                                    # Clear CPU flags
+    zerox                                           # Clear xCPU state
```

Comment:
This works, but dangerously. `zerox` pushes `%rdx` and `%rax` to the stack. But you just changed the position of the stack (`%rsp`) a few lines above. Therefore, when you push to the stack you are overwriting legitimate values and corrupting the stack.

Reply:
I see, `mov  0x48(%rsp), %rsp` set the stack pointer back to where the untrusted nonvolatile register states are saved.